dev/io.openliberty.data.internal_fat_jpa/test-applications/DataJPATestApp/src/test/jakarta/data/jpa/web/DataJPATestServlet.java

```diff
@@ -570,6 +575,98 @@ public void testEmbeddedId() {
         accounts.deleteByOwnerEndsWith("TestEmbeddedId");
     }
 
+    /**
+     * Tests CrudRepository methods that supply entities as parameters.
+     */
+    @Test
+    public void testEntitiesAsParameters() throws Exception {
+        orders.deleteAll();
+
+        Order o1 = new Order();
+        o1.purchasedBy = "testEntitiesAsParameters-Customer1";
+        o1.purchasedOn = OffsetDateTime.now();
+        o1.total = 10.99f;
+        o1 = orders.save(o1);
+        int o1_v1 = o1.versionNum;
+
+        Order o2 = new Order();
+        o2.purchasedBy = "testEntitiesAsParameters-Customer2";
+        o2.purchasedOn = OffsetDateTime.now();
+        o2.total = 20.99f;
+        o2 = orders.save(o2);
+
+        Order o3 = new Order();
+        o3.purchasedBy = "testEntitiesAsParameters-Customer3";
+        o3.purchasedOn = OffsetDateTime.now();
+        o3.total = 30.99f;
+        o3 = orders.save(o3);
+
+        Order o4 = new Order();
+        o4.purchasedBy = "testEntitiesAsParameters-Customer4";
+        o4.purchasedOn = OffsetDateTime.now();
+        o4.total = 40.99f;
+        o4 = orders.save(o4);
+
+        Order o5 = new Order();
+        o5.purchasedBy = "testEntitiesAsParameters-Customer5";
+        o5.purchasedOn = OffsetDateTime.now();
+        o5.total = 50.99f;
+        o5 = orders.save(o5);
+        int o5_v1 = o5.versionNum;
+
+        // delete even though a property doesn't match
+        o4.total = 44.99f;
+        orders.delete(o4);
+
+        // cannot delete when the version number doesn't match
+        tran.begin();
+        try {
+            o1 = orders.findById(o1.id).orElseThrow();
+            Long o1id = o1.id;
+
+            // Update in separate transaction:
+            CompletableFuture.supplyAsync(() -> {
+                Order o1updated = orders.findById(o1id).orElseThrow();
+                o1updated.total = 11.99f;
+                return orders.save(o1updated);
+            }).get(30, TimeUnit.SECONDS);
+
+            try {
+                orders.delete(o1);
+                fail("Deletion must be rejected when the version doesn't match.");
+            } catch (DataException x) {
+                System.out.println("Deletion was rejected as it ought to be be when the version does not match.");
```

Comment:
There's an extra 'be'

Reply:
Thanks for spotting that, it's fixed under https://github.com/OpenLiberty/open-liberty/pull/25336/commits/7feef3d6707c156c0c33212287a656a643f358a4 along with a resolution to the TODO for the IdClass/version combo.